packages/kit/src/runtime/server/page/csp.js

```diff
@@ -0,0 +1,197 @@
+import { escape_html_attr } from '../../../utils/escape.js';
+import { sha256, base64 } from './crypto.js';
+
+/** @type {Promise<void>} */
+export let csp_ready;
+
+/** @type {() => string} */
+let generate_nonce;
+
+/** @type {(input: string) => string} */
+let generate_hash;
+
+if (typeof crypto !== 'undefined') {
+	const array = new Uint8Array(16);
+
+	generate_nonce = () => {
+		crypto.getRandomValues(array);
+		return base64(array);
+	};
+
+	generate_hash = sha256;
+} else {
+	const name = 'crypto'; // store in a variable to fool esbuild when adapters bundle kit
```

Comment:
esbuild evolves and might not be fooled by this forever. could we explicitly tell it to ignore crypto?

Reply:
we could, but every adapter that used esbuild would need to do it separately, including ones we don't maintain. it seemed simpler to solve it here. if esbuild's behaviour changed, we'd catch it in CI (that's how i caught this one)